esphome/components/ltr303/ltr303.h

```diff
@@ -0,0 +1,160 @@
+#pragma once
+
+#include "esphome/components/i2c/i2c.h"
+#include "esphome/components/sensor/sensor.h"
+#include "esphome/core/component.h"
+#include "esphome/core/optional.h"
+
+namespace esphome {
+namespace ltr303 {
+
+// https://www.mouser.com/datasheet/2/239/Lite-On_LTR-303ALS-01_DS_ver%201.1-1175269.pdf
+
+enum CommandRegisters : uint8_t {
+  CR_ALS_CTRL = 0x80,   // ALS operation mode control SW reset
+  CR_MEAS_RATE = 0x85,  // ALS measurement rate in active mode
+  CR_PART_ID = 0x86,    // Part Number ID and Revision ID
+  CR_MANU_ID = 0x87,    // Manufacturer ID
+  CR_CH1_0 = 0x88,      // ALS measurement CH1 data, lower byte - infrared only
+  CR_CH1_1 = 0x89,      // ALS measurement CH1 data, upper byte - infrared only
+  CR_CH0_0 = 0x8A,      // ALS measurement CH0 data, lower byte - visible + infrared
+  CR_CH0_1 = 0x8B,      // ALS measurement CH0 data, upper byte - visible + infrared
+  CR_ALS_STATUS = 0x8c  // ALS new data status
+};
```

Comment:
In ESPHome, we typically use (static) consts for this purpose.
```suggestion
static const uint8_t CR_ALS_CTRL = 0x80;   // ALS operation mode control SW reset
static const uint8_t CR_MEAS_RATE = 0x85;  // ALS measurement rate in active mode
static const uint8_t CR_PART_ID = 0x86;    // Part Number ID and Revision ID
static const uint8_t CR_MANU_ID = 0x87;    // Manufacturer ID
static const uint8_t CR_CH1_0 = 0x88;      // ALS measurement CH1 data, lower byte - infrared only
static const uint8_t CR_CH1_1 = 0x89;      // ALS measurement CH1 data, upper byte - infrared only
static const uint8_t CR_CH0_0 = 0x8A;      // ALS measurement CH0 data, lower byte - visible + infrared
static const uint8_t CR_CH0_1 = 0x8B;      // ALS measurement CH0 data, upper byte - visible + infrared
static const uint8_t CR_ALS_STATUS = 0x8c; // ALS new data status
```

Reply:
yes, i saw this usage in other components, but I think we might move forward small steps at least.
Simple `static const` are really from the old C-style past. 
I believe enums give us much more clearer and readable code, especially if used prefixed with enum name like `RegisterMap::CONF_ABC`. Also it works as typecheck.
If there is one or two values - consts are okay, 
when we have numbers of them - semantically its better to organize them into enums.

I'm strong believer we shall keep with enums, even if is is (yet) look a little bit different than old approach. All in all its not of that big difference in the end, but readability much better.
What do you think? 

I have few more components I in progress of making now. All with enums now, really easier to read.  (VEML7700, INA228).

Reply:
I'm not strictly opposed to it if it's a generally-accepted new(ish) approach -- would be curious if @jesserockz has an opinion. I've seen it come up elsewhere a couple of times (outside of your PRs) in the last few weeks but didn't realize it's a new pattern people are adopting.

Reply:
resolving as 'happy to disagree', moreover a lot of other things changed as well